aya/src/obj/mod.rs

```diff
@@ -670,32 +689,45 @@ mod tests {
                     value_size: 3,
                     max_entries: 4,
                     map_flags: 5,
-                    id: 0,
-                    pinning: 0
+                    ..Default::default()
                 })
             ),
-            Ok(bpf_map_def {
-                map_type: 1,
-                key_size: 2,
-                value_size: 3,
-                max_entries: 4,
-                map_flags: 5,
-                id: 0,
-                pinning: 0
-            })
+            Ok(map_def)
+        ));
+
+        assert!(matches!(
+            parse_map_def(
+                "foo",
+                &[0u8, 0u8, 0u8, 1u8, 0u8, 0u8, 0u8, 2u8, 0u8, 0u8, 0u8, 4u8, 0u8, 0u8, 0u8, 5u8]
+            ),
+            Ok(map_def)
         ));
+        let map = parse_map_def(
+            "foo",
+            &[0u8, 0u8, 0u8, 1u8, 0u8, 0u8, 0u8, 2u8, 0u8, 0u8, 0u8, 4u8, 0u8, 0u8, 0u8, 5u8]
+        ).unwrap();
+        assert!(map.id == 0 && map.pinning == 0)
     }
 
     #[test]
     fn test_parse_map_error() {
         assert!(matches!(
-            parse_map(&fake_section("maps/foo", &[]), "foo"),
+            parse_map(&fake_section("maps/foo", &[0u8; std::mem::size_of::<bpf_map_def>() + 10]), "foo"),
             Err(ParseError::InvalidMapDefinition { .. })
```

Comment:
the old test was correct, please revert

Reply:
Reverted in 1404d12a7e29ab02225bc13fd36794f6c6206db1